Files.Launcher/DeviceWatcher.cs

```diff
@@ -49,6 +54,19 @@ private async void DeviceModifiedEvent(object sender, EventArrivedEventArgs e)
             await SendEvent(deviceName, deviceId, eventType);
         }
 
+        private async void CdModifiedEvent(object sender, EventArrivedEventArgs e)
+        {
+            CimInstance obj = (CimInstance)e.NewEvent.Instance.CimInstanceProperties["TargetInstance"].Value;
+            var deviceName = (string)obj.CimInstanceProperties["Name"]?.Value;
+            var deviceId = (string)obj.CimInstanceProperties["DeviceID"]?.Value;
```

Comment:
To be able to match a drive this should be: `var deviceId = (string)obj.CimInstanceProperties["Drive"]?.Value;`